getspice.py

```diff
@@ -43,116 +41,210 @@
+        ('Darwin', '64bit'): ('MacIntel_OSX_AppleC_64bit', 'tar.Z'),
+        ('cygwin', '32bit'): ('PC_Cygwin_GCC_32bit', 'tar.Z'),
+        ('cygwin', '64bit'): ('PC_Cygwin_GCC_64bit', 'tar.Z'),
+        ('FreeBSD', '32bit'): ('PC_Linux_GCC_32bit', 'tar.Z'),
+        ('FreeBSD', '64bit'): ('PC_Linux_GCC_64bit', 'tar.Z'),
+        ('Linux', '32bit'): ('PC_Linux_GCC_32bit', 'tar.Z'),
+        ('Linux', '64bit'): ('PC_Linux_GCC_64bit', 'tar.Z'),
+        ('Windows', '32bit'): ('PC_Windows_VisualC_32bit', 'zip'),
+        ('Windows', '64bit'): ('PC_Windows_VisualC_64bit', 'zip')}
+
+    def __init__(self, version='N0066'):
+        """Init method that uses either the default N0066 toolkit version token
+        or a user provided one.
+        """
         try:
-            print("Attempting to download spice...")
-            download = downloadSpice(root_url + result)
-            print('Unpacking... (this may take some time!)')
-            if result.endswith('zip'):
-                filelike = io.BytesIO(download.read())
-                with zipfile.ZipFile(filelike, 'r') as archive:
-                    archive.extractall(root_dir)
-                filelike.close()
-            else:
-                cmd = 'gunzip | tar xC ' + root_dir
-                proc = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE)
-                proc.stdin.write(download.read())
-            download.close()
-            break
-        except urllib.error.URLError as err:
-            print("Download failed with URLError: {}, trying again after 15 seconds!".format(err.reason))
-        attempts += 1
-        time.sleep(15)
-
-
-if __name__ == '__main__':
-    # Call getSpice
-    getSpice()
+            # Get the remote file path for the Python architecture that
+            # executes the script.
+            distribution, self._ext = self._distribution_info()
+        except KeyError:
+            print('SpiceyPy currently does not support your system.')
+
+        else:
+            cspice = 'cspice.{}'.format(self._ext)
+            self._rcspice = ('https://naif.jpl.nasa.gov/pub/naif/misc'
+                             '/toolkit_{0}/C/{1}/packages'
+                             '/{2}').format(version, distribution, cspice)
+
+            # Setup the local directory (where the package will be downloaded)
+            self._root = os.path.realpath(os.path.dirname(__file__))
+
+            # Download the file
+            print('Downloading CSPICE for {0}...'.format(distribution))
+            attempts = 10   # Let's try a maximum of attempts for getting SPICE
+            while attempts:
+                attempts -= 1
+                try:
+                    self._download()
+                except RuntimeError as error:
+                    print("Download failed with URLError: {0}, trying again after "
+                          "15 seconds!".format(error))
+                else:
+
```

Comment:
there are a lot of new lines... its a style thing but for me this is too sparse.